City-of-Helsinki/drupal-helfi-rekry
# Uhf 5254: Employment dropdown

# [UHF-5254](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5254)
Add dropdown for employment options + backend logic for indexing.

## What was done
- Add jotai store for combined employment / employment type options (with predefined list of options)
- Make initial aggs request into `_msearch` request that gets bot aggs and queries indexed taxonomy terms
- Add logic for rendering dropdown / filter pills
- Pass url params correctly
- Add forming query from selections
## How to install

Make sure your instance is up and running on correct branch.
  * `git fetch; git checkout origin/UHF-5254-employment-dropdown`
  * `make fresh` OR `make drush-cr; make drush-cim;`
* Run `make drush-cr`

Set up your index: 
Env;
* Run make ps. Note the port that your drupal-helfi-rekry-elastic container is running in.
* Add :point_up: to your .env as the ELASTIC_PROXY_URL variable (http:localhost:{port-elastic-runs-in})
* Run make up
* make shell into the container. Verify that your ELASTIC_PROXY_URL is set by running printenv | grep ELASTIC_PROXY_URL

Elastic:
* If you have set up index before, it might be a good idea to delete it beforehand with curl -X DELETE http://localhost/:{port-elastic-runs-in}/job_listings
* Run drush sapi-rt && drush sapi-i
* curl -i http://localhost/:{port-elastic-runs-in}/job_listings to verify that the index is up and running (you should get a response with info about the cluster)

## How to test
Frontpage form: 
 * Add a `job_search` paragraph to a landing page with value in `field_job_search_result_page` (Avoimet työpaikat node as the value)
 * View the page. It should look like this: https://share.goabstract.com/060a4426-0870-4e29-ac48-006d68293bf6?collectionLayerId=694b53e0-8010-4798-88cc-ebda1b339ab6&mode=design&sha=latest
 * Adding values to fields and submitting should redirect to the full search results page and prefill selections based url params

Full search page: 
* Making selections in the employment dropdown should correctly filter the results and update url params
* Should look like this: https://share.goabstract.com/060a4426-0870-4e29-ac48-006d68293bf6?collectionLayerId=a68e8006-2091-4eb7-a570-ad85540bc5a2&mode=design&sha=latest

## Designers review
